pkg/quota/admission/runonceduration/admission_test.go

```diff
@@ -0,0 +1,162 @@
+package runonceduration
+
+import (
+	"bytes"
+	"testing"
+
+	"k8s.io/kubernetes/pkg/admission"
+	kapi "k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/client/cache"
+	ktestclient "k8s.io/kubernetes/pkg/client/unversioned/testclient"
+
+	oadmission "github.com/openshift/origin/pkg/cmd/server/admission"
+	projectcache "github.com/openshift/origin/pkg/project/cache"
+)
+
+func testCache() *projectcache.ProjectCache {
+	kclient := &ktestclient.Fake{}
+	pCache := projectcache.NewFake(kclient.Namespaces(), projectcache.NewCacheStore(cache.MetaNamespaceKeyFunc), "")
+	ns := &kapi.Namespace{}
+	ns.Name = "default"
+	ns.Annotations = map[string]string{
+		"openshift.io/active-deadline-seconds-override": "1000",
+	}
+	pCache.Store.Add(ns)
+	return pCache
+}
+
+func testConfig(n *int64, annotationOverride bool) *RunOnceDurationConfig {
+	return &RunOnceDurationConfig{
+		ActiveDeadlineSecondsOverride:  n,
+		AllowProjectAnnotationOverride: annotationOverride,
+	}
+}
+
+func testRunOncePod() *kapi.Pod {
+	pod := &kapi.Pod{}
+	pod.Spec.RestartPolicy = kapi.RestartPolicyNever
+	return pod
+}
+
+func testRestartOnFailurePod() *kapi.Pod {
+	pod := &kapi.Pod{}
+	pod.Spec.RestartPolicy = kapi.RestartPolicyOnFailure
+	return pod
+}
+
+func testRunOncePodWithDuration(n int64) *kapi.Pod {
+	pod := testRunOncePod()
+	pod.Spec.ActiveDeadlineSeconds = &n
+	return pod
+}
+
+func testRestartAlwaysPod() *kapi.Pod {
+	pod := &kapi.Pod{}
+	pod.Spec.RestartPolicy = kapi.RestartPolicyAlways
+	return pod
+}
+
+func int64p(n int64) *int64 {
+	return &n
+}
+
+func TestRunOnceDurationAdmit(t *testing.T) {
+	tests := []struct {
+		config                        *RunOnceDurationConfig
+		pod                           *kapi.Pod
+		expectedActiveDeadlineSeconds *int64
+	}{
+		// 0: expect globally configured duration to be set
```

Comment:
Style suggestion: give your test structs a string name/label field, and make their value what you current have as comments; then in your test processing loop, log the name of each test as it's evaluated. Then when there's a failure it's much more clear which test array entry is associated with the failure.